yewtil/src/not_equal_assign.rs

```diff
@@ -43,13 +43,74 @@ pub trait NeqAssign<NEW> {
     ///#         unimplemented!()
     ///#     }
     ///  }
+    ///
+    /// let mut foo = 1;
+    ///
+    /// assert_eq!(foo.neq_assign(42), true);
+    /// assert_eq!(foo, 42);
+    ///
+    /// assert_eq!(foo.neq_assign(42), false);
     /// ```
     fn neq_assign(&mut self, new: NEW) -> ShouldRender;
 }
 
 impl<T: BorrowMut<U>, U: PartialEq> NeqAssign<U> for T {
     fn neq_assign(&mut self, new: U) -> bool {
-        if self.borrow() != &new {
+        self.neq_assign_by(new, |x, y| x == y)
```

Comment:
While I appreciate the code reuse here, I'm not 100% sure that passing a `FnOnce` closure here will be optimized to the same emitted WASM as just doing the comparison in the same way as before.

I'd like to have some evidence that the trait fn call and closure are inlined, but I understand that verifying that via inspecting emitted WASM can be a difficult and time-consuming task, so feel free not to. :man_shrugging:

Reply:
Unfortunately, I don't currently even know where to begin with how to do that. I would like to learn at some point, though.

I did ask myself this question, though. I looked at implementations of `*_by` functions in `std` (e.g. [`Iterator::eq_by`](https://doc.rust-lang.org/src/core/iter/traits/iterator.rs.html#3005-3013)), and `eq` is implemented in terms of `eq_by`. I didn't see any `#[inline]` attributes either. I could be wrong, but I think this would be an obvious compiler optimization.